test/e2e/apimachinery/aggregator.go

```diff
@@ -550,6 +576,368 @@ func TestSampleAPIServer(f *framework.Framework, aggrclient *aggregatorclient.Cl
+			options.LabelSelector = "apiservice=created"
+			return apiServiceClient.Watch(context.TODO(), options)
+		},
+	}
+
+	apiServiceList, err := apiServiceClient.List(context.TODO(), metav1.ListOptions{LabelSelector: "apiservice=created"})
+	framework.ExpectNoError(err, "failed to list API Services")
+
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		statusToUpdate, err = apiServiceClient.Get(context.TODO(), apiServiceName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "Unable to retrieve api service %s", apiServiceName)
+
+		statusToUpdate.Status.Conditions = append(statusToUpdate.Status.Conditions, apiregistrationv1.APIServiceCondition{
+			Type:    "StatusUpdate",
+			Status:  "True",
+			Reason:  "E2E",
+			Message: "Set from e2e test",
+		})
+
+		updatedStatus, err = apiServiceClient.UpdateStatus(context.TODO(), statusToUpdate, metav1.UpdateOptions{})
+		return err
+	})
+	framework.ExpectNoError(err, "Failed to update status. %v", err)
+	framework.Logf("updatedStatus.Conditions: %#v", updatedStatus.Status.Conditions)
+
+	ginkgo.By("watching for the APIService to be updated")
+	ctx, cancel := context.WithTimeout(context.Background(), apiServiceRetryTimeout)
+	defer cancel()
+	_, err = watchtools.Until(ctx, apiServiceList.ResourceVersion, w, func(event watch.Event) (bool, error) {
+		if resource, ok := event.Object.(*apiregistrationv1.APIService); ok {
+			found := resource.ObjectMeta.Name == apiServiceName &&
+				resource.Labels["apiservice"] == "created"
+			if !found {
+				framework.Logf("Observed APIService %v with Labels: %v & Conditions: %v", resource.ObjectMeta.Name, resource.Labels, resource.Status.Conditions)
+				return false, nil
+			}
+			for _, cond := range resource.Status.Conditions {
+				if cond.Type == "StatusUpdate" &&
+					cond.Reason == "E2E" &&
+					cond.Message == "Set from e2e test" {
+					framework.Logf("Found APIService %v with Labels: %v & Conditions: %v", resource.ObjectMeta.Name, resource.Labels, cond)
+					return found, nil
+				} else {
+					framework.Logf("Observed APIService %v with Labels: %v & Conditions: %v", resource.ObjectMeta.Name, resource.Labels, cond)
+				}
+			}
+			return false, nil
+		}
+		framework.Logf("Observed event: %+v", event.Object)
+		return false, nil
+	})
+	framework.ExpectNoError(err, "failed to locate APIService %v with conditions: %v", apiServiceName, updatedStatus.Status.Conditions)
+	framework.Logf("APIService Status for %s has been updated", apiServiceName)
+
+	ginkgo.By("Delete a collection of APIServices")
+	one := int64(1)
+	err = aggrclient.ApiregistrationV1().APIServices().DeleteCollection(context.TODO(),
+		metav1.DeleteOptions{GracePeriodSeconds: &one},
+		metav1.ListOptions{LabelSelector: "apiservice=created"})
+	framework.ExpectNoError(err, "Unable to delete apiservice %s", apiServiceName)
+	framework.Logf("APIService %s has been deleted.", apiServiceName)
+
+	cleanBetaTest(client, aggrclient, namespace)
```

Comment:
Aren't we running this in `AfterEach`?

Reply:
Lines 935 & 936 are checking/logging that the `deleteCollection` has worked successfully

Line 938 was just following the same pattern used in the other [APIService test](https://github.com/ii/kubernetes/blob/1d888b3b88c77f4362c0ebb284967524a3f6e18f/test/e2e/apimachinery/aggregator.go#L576)
```
cleanTest(client, aggrclient, namespace)
```